### PR TITLE
Pass NA in InspectionReport if some entities are null

### PR DESCRIPTION
**Pull request by UsamaEquinorAFK**

### CODE REVIEW

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs [hunk 1]
```diff
@@ -77,19 +81,32 @@ private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> i
...
 
-                if (task?.TagId != null && missionRun?.Area?.Plant?.Name != null && missionRun?.Area?.Name != null)
+                if (task != null && missionRun != null && missionRun?.Area != null)
                 {
                     var finding = new Finding(
-                        task.TagId,
-                        missionRun.Area.Plant.Name,
-                        missionRun.Area.Name,
+                        task.TagId ?? "NA",
+                        missionRun.Area.Plant.Name ?? "NA",
```
Comment (Afonso-2403) [inline backend/api/EventHandlers/InspectionFindingEventHandler.cs:88]:
```suggestion
                        missionRun.Area?.Plant.Name ?? "NA",
```

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs [hunk 2]
```diff
@@ -77,19 +81,32 @@ private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> i
...
+                        missionRun.Area.Name ?? "NA",
```
Comment (Afonso-2403) [inline backend/api/EventHandlers/InspectionFindingEventHandler.cs:89]:
```suggestion
                        missionRun.Area?.Name ?? "NA",
```

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs [hunk 3]
```diff
@@ -77,19 +81,32 @@ private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> i
...
                         inspectionFinding.Finding,
                         inspectionFinding.InspectionDate,
-                        missionRun.Robot.Name
+                        missionRun.Robot.Name ?? "NA"
                     );
 
                     findingsList.Add(finding);
                 }
+                else if (task != null && missionRun?.Area == null && missionRun != null)
```
Comment (Afonso-2403) [inline backend/api/EventHandlers/InspectionFindingEventHandler.cs:97]: With the suggestions above I think you can drop this else

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs [hunk 4]
```diff
@@ -77,19 +81,32 @@ private async Task<List<Finding>> GenerateFindingsList(List<InspectionFinding> i
                 var missionRun = await InspectionFindingService.GetMissionRunByIsarStepId(inspectionFinding);
                 var task = await InspectionFindingService.GetMissionTaskByIsarStepId(inspectionFinding);
 
-                if (task?.TagId != null && missionRun?.Area?.Plant?.Name != null && missionRun?.Area?.Name != null)
+                if (task != null && missionRun != null && missionRun?.Area != null)
```
Comment (Afonso-2403) [inline backend/api/EventHandlers/InspectionFindingEventHandler.cs:84]:
```suggestion
                if (task != null && missionRun != null)
```

Comment (UsamaEquinorAFK) [reply backend/api/EventHandlers/InspectionFindingEventHandler.cs:84]: Agreed, done.

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs (hunk 2 above)
Comment (UsamaEquinorAFK) [reply backend/api/EventHandlers/InspectionFindingEventHandler.cs:89]: Agreed, done.

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs (hunk 1 above)
Comment (UsamaEquinorAFK) [reply backend/api/EventHandlers/InspectionFindingEventHandler.cs:88]: Agreed, done.

File: backend/api/EventHandlers/InspectionFindingEventHandler.cs (hunk 3 above)
Comment (UsamaEquinorAFK) [reply backend/api/EventHandlers/InspectionFindingEventHandler.cs:97]: Agreed, done.

**Review by andchiind — APPROVED** [review]
This does imply we cannot have a plant/area/robot with the name "NA", so I'm not sure if this is ideal compared to a longer string that says that it's missing, or making the fields nullable in Finding. But for now we can leave it like this for testing

**Comment by UsamaEquinorAFK** [comment]
> This does imply we cannot have a plant/area/robot with the name "NA", so I'm not sure if this is ideal compared to a longer string that says that it's missing, or making the fields nullable in Finding. But for now we can leave it like this for testing

I agree. What we mean by NA here is Not Available. Further more we need to review the "required" status of various attributes in the database models. For example why would TagID ever be nullable in MissionTask same goes for Area in MissionRun.

**Pull request merged**